src/shared/components/toggle-button/index.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react'
+import classnames from 'classnames'
+import PropTypes from 'prop-types'
+
+import styles from './index.module.css'
+
+class ToggleButton extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      isActive: props.initialState || false
+    }
```

Comment:
What is the logic that you intend to use for the service worker enabling/disabling? That is, Will you have another component which enables/disables the service worker and will use this component? The service worker here is an example, it could be anything.

If those operations are intended to be outside of the scope of this toggle component (which I think to be the best solution since you would have a generic component here), you would not need this state and you should have an `isActive` prop, as well as an `onToggleChange` prop.

Reply:
Yes, that's the logic that I'm trying to use here: a parent component that will use and control `toggle-button` component. 
I understood and agree with you. I'll change update it now. Thanks.